compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/internal/FunctionParamComparator.java

```diff
@@ -30,8 +31,13 @@ public class FunctionParamComparator implements Comparator<BIRNode.BIRVariableDc
 
     @Override
     public int compare(BIRNode.BIRVariableDcl o1, BIRNode.BIRVariableDcl o2) {
-
-        return Integer.compare(getWeight(o1), getWeight(o2));
+        if (o1.kind != o2.kind) {
+            return Integer.compare(getWeight(o1), getWeight(o2));
+        }
+        if (o1.kind == VarKind.TEMP) {
+            return Integer.compare(o1.name.value.hashCode(), o2.name.value.hashCode());
```

Comment:
Shall we modify the above javadoc as follows. `BIR Function Param comparator to sort according to RETURN, ARG varkinds and TEMP vars according to their names.`